tiny_dnn/util/serialization_functions.h

```diff
@@ -47,75 +47,6 @@ void arc(Archive &ar, cereal::NameValuePair<size_t> &&arg) {
   arg.value = arg2.value;
 }
 
-static serial_size_t to_serial(const size_t &src) {
```

Comment:
I confirmed they aren't actually needed. Only `serial_size_t` <=> `size_t` conversion wrapped in `cereal::NameValuePair` type is enough. 

With other arc functions and this function
```cpp
template <class Archive, typename T>
void serialize(Archive &ar, tiny_dnn::index3d<T> &idx) {
  arc(ar, make_nvp("width", idx.width_), make_nvp("height", idx.height_),
      make_nvp("depth", idx.depth_));
}
```
cereal library can do the job with some horrible code generation.